app/src/components/datasets/DatasetEntriesTable/DatasetEntryEditorDrawer.tsx

```diff
@@ -57,6 +58,15 @@ export default function DatasetEntryEditorDrawer({
     }
   }, [savedInputMessages, savedOutputMessage]);
 
+  const initialHash = useMemo(
```

Comment:
I might be missing something but this feels unnecessarily convoluted/difficult to follow? Seems like we could get the same effect in a more straightforward way with something like:

```ts
import { isEqual } from 'lodash-es';
// ...

const hasUpdates = useMemo(
  () =>
    !isEqual(datasetEntry?.input, inputMessagesToSave) ||
    !isEqual(datasetEntry?.output, outputMessageToSave),
  [datasetEntry, inputMessagesToSave, outputMessageToSave],
);
```

Reply:
Yep, forgot about isEqual